market/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+//! # Cryptocurrency market pallet
+//!
+//! The Market pallet provides functionality for handling exchange rates of the coin (ex: TEER) to different currencies
+//!
+//! - [`Config`]
+//! - [`Call`]
+//! - [`Pallet`]
+//!
+//! ## Overview
+//!
+//! The Market pallet provides functions for:
+//!
+//! - Setting exchange rates.
+//!
+//!
+#![cfg_attr(not(feature = "std"), no_std)]
+pub use pallet::*;
+pub use substrate_fixed::types::U32F32;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::weights::WeightInfo;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_std::prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_teerex::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn exchange_rate)]
+	pub(super) type ExchangeRates<T> = StorageMap<_, Blake2_128Concat, Vec<u8>, U32F32, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The exchange rate of currency was set/updated. \[currency], [new value\]
+		ExchangeRateUpdated(Vec<u8>, Option<U32F32>),
```

Comment:
Update of?

Reply:
update of all the substrate deps, but this is ok.

Reply:
Do you mean, solving task https://github.com/integritee-network/integritee-node/issues/83 ? Because the upgrade of the node required also an update of the pallets.

Reply:
yes.